web/frontend/src/components/slam-2d-render.vue

```diff
@@ -110,6 +166,20 @@ const updateCloud = (pointcloud: Uint8Array) => {
   intersectionPlane.position.set(center.x, 0, center.z);
   raycaster.objects = [intersectionPlane];
 
+  const colors = points.geometry.attributes.color;
+  // if the PCD has a color attribute defined, convert those colors using the colorMap
+  if (colors instanceof THREE.BufferAttribute || colors instanceof THREE.InterleavedBufferAttribute) {
```

Comment:
looks like it could be an instance of an interleavedBufferAttribute if colors had other fields packed into it. From looking at parse it does not seem like we can get an interleavedBufferAttribute, and I cannot comment if the behavior will be the same as a BufferAttribute.

Should I go ahead and remove it from the if statement?

Reply:
Yeah, I don't see how it could be InterleavedBufferAttribute from PCDLoader either. I think we should remove it from the if statement.

For my info, what colors get displayed if `points.geometry.attributes.color` doesn't exist?

Reply:
if `color` does not exist you end up with a very faded output in the 2D view. taken from orbslam
<img width="648" alt="Screenshot 2023-04-10 at 4 31 50 PM" src="https://user-images.githubusercontent.com/92045055/230992747-29019b04-0847-4b30-961d-20fba9dc360c.png">

the 3D viewer still looks fine visually, but does not include a marker.
<img width="957" alt="Screenshot 2023-04-10 at 4 33 55 PM" src="https://user-images.githubusercontent.com/92045055/230993014-b2ec2a94-804d-49af-8ac9-258f0992f95c.png">

Reply:
Perfect, thanks for explaining!